drivers/platform/maxim/gpio_extra.h

```diff
@@ -57,25 +57,12 @@ extern const struct gpio_platform_ops max_gpio_ops;
  */
 extern const struct irq_platform_ops max_gpio_irq_ops;
 
-/**
- * @struct gpio_irq_config
- * @brief maxim configuration descriptor for irq operations
- */
-struct gpio_irq_config {
-	enum irq_trig_level mode;
-};
-
 struct max_gpio_init_param {
 	/** Input/Output */
 	uint32_t mode;
 	/** Pull_up/Pull_down  */
-	uint32_t pull;
+	enum gpio_pull_up pull;
```

Comment:
Since enum gpio_pull_up pull is generic, it should go in gpio_init_param and gpio_desc, not here.
But I'm wondering now, if it'd be better to create a function for this because this setting is only applicable to inputs.